bioptim/interfaces/acados_interface.py

```diff
@@ -87,12 +98,47 @@ def __set_constr_type(self, constr_type="BGH"):
         self.acados_ocp.constraints.constr_type = constr_type
         self.acados_ocp.constraints.constr_type_e = constr_type
 
+    def __dispatch_bounds(self, ocp):
+        all_g_bounds = Bounds(interpolation=InterpolationType.CONSTANT)
+        for i in range(ocp.nb_phases):
+            for g, G in enumerate(ocp.nlp[i].g):
+                all_g_bounds.concatenate(G[0]["bounds"])
+
+        if isinstance(all_g_bounds.min, (SX, MX)) or isinstance(all_g_bounds.max, (SX, MX)):
+            raise RuntimeError("Ipopt doesn't support SX/MX types in constraints bounds")
+        return all_g_bounds
+
     def __set_constrs(self, ocp):
         # constraints handling in self.acados_ocp
         u_min = np.array(ocp.nlp[0].u_bounds.min)
         u_max = np.array(ocp.nlp[0].u_bounds.max)
         x_min = np.array(ocp.nlp[0].x_bounds.min)
         x_max = np.array(ocp.nlp[0].x_bounds.max)
+        self.all_constr = SX()
+        self.end_constr = SX()
+        ##TODO:change for more node flexibility on bounds
+        self.all_g_bounds = Bounds(interpolation=InterpolationType.CONSTANT)
+        self.end_g_bounds = Bounds(interpolation=InterpolationType.CONSTANT)
+        for i in range(ocp.nb_phases):
+            for g, G in enumerate(ocp.nlp[i].g):
+                if G and G[0]["constraint"].node[0] is Node.ALL:
+                    self.all_constr = vertcat(self.all_constr, G[0]["val"].reshape((-1, 1)))
+                    self.all_g_bounds.concatenate(G[0]["bounds"])
+                    if len(G) > ocp.nlp[0].ns:
+                        constr_end_func_tp = Function(f"cas_constr_end_func_{i}_{g}", [ocp.nlp[i].X[-1]], [G[0]["val"]])
+                        self.end_constr = vertcat(self.end_constr, constr_end_func_tp(ocp.nlp[i].X[0]).reshape((-1, 1)))
+                        self.end_g_bounds.concatenate(G[0]["bounds"])
+
+                elif G and G[0]["constraint"].node[0] is Node.END:
+                    constr_end_func_tp = Function(f"cas_constr_end_func_{i}_{g}", [ocp.nlp[i].X[-1]], [G[0]["val"]])
+                    self.end_constr = vertcat(self.end_constr, constr_end_func_tp(ocp.nlp[i].X[0]).reshape((-1, 1)))
+                    self.end_g_bounds.concatenate(G[0]["bounds"])
+
+                elif G and G[0]["constraint"].node[0] is Node.START:
+                    raise RuntimeError("Acados solver only handles constraints on first node for states and controls.")
```

Comment:
If the user specified Node.INTERMIDIATE, it will silently be ignored. Please use an `else` somewhere to make sure all the cases are properly treated

Reply:
OK